pkg/api/validation/validation.go

```diff
@@ -646,7 +671,7 @@ func validateEnvVarValueFrom(ev api.EnvVar) errs.ValidationErrorList {
 	switch {
 	case ev.ValueFrom.FieldRef != nil:
 		numSources++
-		allErrs = append(allErrs, validateObjectFieldSelector(ev.ValueFrom.FieldRef).Prefix("fieldRef")...)
+		allErrs = append(allErrs, validateObjectFieldSelector(ev.ValueFrom.FieldRef, &validFieldPathExpressions).Prefix("fieldRef")...)
```

Comment:
I take it back, that's exactly what you did.  Not sure I like it, but I guess we have not defined how labels/annotations would convert into env vars.  Any other reasons?